lib/matplotlib/ticker.py

```diff
@@ -1170,25 +1170,188 @@ class LogitFormatter(Formatter):
+
+    def set_minor_threshold(self, minor_threshold):
+        """
+        Set the threshold for labelling minors ticks
+
+        Parameters
+        ----------
+        minor_threshold : int
+            maximum number of locs for labelling some minor ticks. This
+            parameter have no effect if minor is False.
+        """
+        self._minor_threshold = minor_threshold
+
+    def set_minor_number(self, minor_number):
+        """
+        Set the number of minor ticks to label when some minor ticks are
+        labelled.
+
+        Parameters
+        ----------
+        minor_number : int
+            number of ticks which are labelled when the number of ticks is
+            below the threshold.
+        """
+        self._minor_number = minor_number
+
+    def set_locs(self, locs):
+        self.locs = locs
+        self._labelled.clear()
+
+        if not self._minor:
+            return None
+        if all(
+            is_decade(x, tol=1e-7)
+            or is_decade(1 - x, tol=1e-7)
+            or (is_close_to_int(2 * x) and int(np.round(2 * x)) == 1)
+            for x in locs
+        ):
+            # minor ticks are subsample from ideal, so no label
+            return None
+        if len(locs) < self._minor_threshold:
+            if len(locs) < self._minor_number:
+                self._labelled.update(locs)
+            else:
+                alocs = np.array(locs)
+                diff = np.diff(-np.log(1 / alocs - 1))
+                dnext = np.concatenate(((0,), diff))
+                dprev = np.concatenate((diff, (0,)))
+                criterion = np.min((dnext, dprev), axis=0) + 0.1 * (
+                    dnext + dprev
+                )
+                good_minor = np.argsort(criterion)[-self._minor_number:]
+                self._labelled.update(locs[i] for i in good_minor)
+
     def __call__(self, x, pos=None):
-        s = ''
-        if 0.01 <= x <= 0.99:
-            s = '{:.2f}'.format(x)
-        elif x < 0.01:
-            if is_decade(x):
-                s = '$10^{{{:.0f}}}$'.format(np.log10(x))
+        if self._minor and x not in self._labelled:
+            return ""
+        if x <= 0 or x >= 1:
+            return ""
+        usetex = rcParams["text.usetex"]
+
+        if is_close_to_int(2 * x) and round(2 * x) == 1:
+            s = r"\frac{1}{2}"
+        elif x < 0.5 and is_decade(x, tol=1e-7):
+            exponent = round(np.log10(x))
+            s = "10^{%d}" % exponent
+        elif x > 0.5 and is_decade(1 - x, tol=1e-7):
+            exponent = round(np.log10(1 - x))
+            s = "10^{%d}" % exponent
+            if self._use_overline:
+                s = r"\overline{%s}" % s
             else:
-                s = '${:.5f}$'.format(x)
-        else:  # x > 0.99
-            if is_decade(1-x):
-                s = '$1-10^{{{:.0f}}}$'.format(np.log10(1-x))
+                s = r"1-%s" % s
+        elif x < 0.1:
+            exponent = int(np.floor(np.log10(x)))
```

Comment:
you could use math.floor (here and below), which returns an int, to avoid the additional int() call.

Reply:
done. Thanks.